src/crypto/rsa_key_pair.cpp

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+
+#include "rsa_key_pair.h"
+
+#include "mbedtls/rsa_key_pair.h"
+#include "openssl/rsa_key_pair.h"
+
+namespace crypto
+{
+#ifdef CRYPTO_PROVIDER_IS_MBEDTLS
+  using RSAPublicKeyImpl = RSAPublicKey_mbedTLS;
+  using RSAKeyPairImpl = RSAKeyPair_mbedTLS;
+#else
+  using RSAPublicKeyImpl = RSAPublicKey_OpenSSL;
+  using RSAKeyPairImpl = RSAKeyPair_OpenSSL;
+#endif
+
+  RSAPublicKeyPtr make_rsa_public_key(const Pem& public_pem)
+  {
+    return make_rsa_public_key(public_pem.data(), public_pem.size());
+  }
+
+  RSAPublicKeyPtr make_rsa_public_key(const std::vector<uint8_t>& der)
+  {
+    return std::make_shared<RSAPublicKeyImpl>(der);
+  }
+
+  RSAPublicKeyPtr make_rsa_public_key(const uint8_t* data, size_t size)
+  {
+    if (size < 10 || strncmp("-----BEGIN", (char*)data, 10) != 0)
```

Comment:
It'd be good to have the string a constexpr, the len as a name derived constexpr rather than literals inline.

Reply:
Sure, np. What do mean by "name derived"?

Reply:
BEGIN and BEGIN_LEN for example? Looking into it, it seems that strlen(literal) is optimized out anyway.

Reply:
![image](https://user-images.githubusercontent.com/4016369/109174549-d948a980-777c-11eb-8b32-3e23bf993123.png)

Reply:
![image](https://user-images.githubusercontent.com/4016369/109174707-fda48600-777c-11eb-8771-a6d35bcf0c72.png)

Reply:
Oh sure, the names are fine. In the meanwhile I found out that there is `std::char_traits<char>::length(...);` for this and similar purposes now and it seems to work.